packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -557,70 +561,28 @@ void main() {
+    final Rect fabRect = tester.getRect(find.byType(FloatingActionButton));
 
-    expect(floatingActionButtonOriginBottomCenter.dy > floatingActionButtonBottomCenter.dy, true);
-    expect(snackBarTopCenter.dy > floatingActionButtonBottomCenter.dy, true);
-  });
+    // FAB should shift upwards after SnackBar appears.
+    expect(originalFabRect.center.dy > fabRect.center.dy, true);
 
-  testWidgets('Floating SnackBar button text alignment', (WidgetTester tester) async {
-    await tester.pumpWidget(MaterialApp(
-      theme: ThemeData(
-        snackBarTheme: const SnackBarThemeData(behavior: SnackBarBehavior.floating,)
-      ),
-      home: MediaQuery(
-        data: const MediaQueryData(
-          padding: EdgeInsets.only(
-            left: 10.0,
-            top: 20.0,
-            right: 30.0,
-            bottom: 40.0,
-          ),
-        ),
-        child: Scaffold(
-          body: Builder(
-            builder: (BuildContext context) {
-              return GestureDetector(
-                onTap: () {
-                  Scaffold.of(context).showSnackBar(SnackBar(
-                    content: const Text('I am a snack bar.'),
-                    duration: const Duration(seconds: 2),
-                    action: SnackBarAction(label: 'ACTION', onPressed: () {}),
-                  ));
-                },
-                child: const Text('X'),
-              );
-            }
-          ),
-        ),
-      ),
-    ));
-    await tester.tap(find.text('X'));
-    await tester.pump(); // start animation
-    await tester.pump(const Duration(milliseconds: 750)); // Animation last frame.
+    final Offset snackBarTopRight = tester.getTopRight(find.byType(SnackBar));
 
-    final Offset textBottomLeft = tester.getBottomLeft(find.text('I am a snack bar.'));
-    final Offset textBottomRight = tester.getBottomRight(find.text('I am a snack bar.'));
-    final Offset actionTextBottomLeft = tester.getBottomLeft(find.text('ACTION'));
-    final Offset actionTextBottomRight = tester.getBottomRight(find.text('ACTION'));
-    final Offset snackBarBottomLeft = tester.getBottomLeft(find.byType(SnackBar));
-    final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+    // value was derived from floating_action_button_location.dart
+    const int defaultFabPadding = 16;
 
-    expect(textBottomLeft.dx - snackBarBottomLeft.dx, 31.0 + 10.0); // margin + left padding
-    expect(snackBarBottomLeft.dy - textBottomLeft.dy, 27.0); // margin (with no bottom padding)
-    expect(actionTextBottomLeft.dx - textBottomRight.dx, 16.0);
-    expect(snackBarBottomRight.dx - actionTextBottomRight.dx, 31.0 + 30.0); // margin + right padding
-    expect(snackBarBottomRight.dy - actionTextBottomRight.dy, 27.0); // margin (with no bottom padding)
-  }, skip: isBrowser);
+    // FAB should be positioned above the SnackBar by the default padding.
+    expect(snackBarTopRight.dy - defaultFabPadding, fabRect.bottomRight.dy);
```

Comment:
the order is `expect(actual, expected)`, so since `fabRect` is the actual Rect, it should appear first
```suggestion
    expect(fabRect.bottomRight.dy, snackBarTopRight.dy - defaultFabPadding);
```

Reply:
Fixed